### ISSUE TITLE
`helm package` overwrites previous

### ISSUE THREAD

**Issue by nitrag** [helm/helm#5921]
`helm version`: `2.14`

I am attempting to create a CD pipeline for Helm to deploy a private registry to AWS S3.

My directory structure is as follows:
```
/
├── charts/
│   ├── chart1/
│       ├── ...
|    ├── chart2/
│       ├── ...
├── dist/
│   ├── chart1-0.1.0.tgz
│   └── chart1-0.2.0.tgz
│   └── chart2-0.1.0.tgz
└── index.yaml
```

I have a deploy script as follows:
```bash
echo "Packaging chart updates"
helm package -d dist/ charts/chart1
helm package -d dist/ charts/chart2
#add additional charts here

# Update index
helm repo index .

echo "Syncing to S3"
aws s3 sync . s3://....
```

**Concern**: `helm package` will overwrite **existing** _packaged_ versions, changing the `digest` in `index.yaml` (after `helm repo index .`).

Is this a bug? Would the `digest` change have any effect on gitops style orchestrators? And/or my expectation would be if nothing changed in `charts/charts1` then the `digest` also wouldn't change but perhaps the digest is based off a timestamp as well as the contents? 

TLDR: Should `helm package` check for existing?

**Comment by bacongobbler** [comment]
Working as intended. The user knows best in this situation. If they want to override the packaged version, that's up to them. Other third-party packages like chartmuseum and harbor can take alternative opinions on this approach, as this is just an opinion of `helm repo index`.

Closing as intended.
